src/main/java/org/gridsuite/modification/server/service/BuildWorkerService.java

```diff
@@ -137,23 +142,14 @@ public Consumer<Message<String>> consumeBuild() {
     @Bean
     public Consumer<Message<String>> consumeCancelBuild() {
         return message -> {
-            try {
-                BuildCancelContext cancelContext = BuildCancelContext.fromMessage(message);
-
-                if (buildRequests.contains(cancelContext.getReceiver())) {
-                    cancelBuildRequests.put(cancelContext.getReceiver(), cancelContext);
-                }
-
-                // find the completableFuture associated with receiver
-                CompletableFuture<List<ModificationInfos>> future = futures.get(cancelContext.getReceiver());
-                if (future != null) {
-                    future.cancel(true);  // cancel build in progress
-
-                    stoppedPublisherService.publishCancel(cancelContext.getReceiver());
-                    LOGGER.info(CANCEL_MESSAGE + " (receiver='{}')", cancelContext.getReceiver());
-                }
-            } catch (Exception e) {
-                LOGGER.error("Exception in consumeCancelBuild");
+            BuildCancelContext cancelContext = BuildCancelContext.fromMessage(message);
+            cancelBuildRequests.put(cancelContext.getReceiver(), cancelContext);
```

Comment:
I can see one potential bug here, some user could cancel a build that is not running yet and populate the map
If we run a build with the same receiver any time after, (it could be some hours), it would be canceled